05_fft.md

```diff
@@ -0,0 +1,239 @@
+$$
+![Example series plot](./plots/fft_ex_fun2.png){ width=50% }
+
+## Fourier sine series
+A $\sin$ series with coefficients $a_n$ defines a function over a finite interval
+$$
+  f(x) = \sum_n^\infty a_n \sin\left(\frac{n \pi x}{l}\right)\text{,}
+  \quad 0 < x < l \text{.}
+$$
+![Example series plot](./plots/fft_ex_fun2_decomp.png){ width=50% }
+
+## Fourier sine series
+A nice property of the $\sin$ functions is *orthogonality*:
+$$
+  \int_0^l dx \sin\left(\frac{n \pi x}{l}\right)
+  \sin\left(\frac{m \pi x}{l}\right) = 0\text{,} \quad m \neq n\text {.}
+$$
+
+## Fourier sine series
+The orthogonality property makes it easy to compute the coefficients of the series:
+$$
+  \int_0^ldx \sin\left(\frac{n \pi x}{l}\right) f(x)
+  = \int_0^ldx \sin\left(\frac{n \pi x}{l}\right)
+  \sum_m^\infty a_m \sin\left(\frac{m \pi x}{l}\right)
+  = \frac{l}{2} a_n \text{.}
+$$
+
+## Fourier sine series
+It can be shown that for *any* function, the sine series with coefficients
+computed this way converges to that function. This is called *completeness*.
+It means we can decompose any function into a sum of sines.
+![Example series plot](./plots/fft_ex_fun_decomp.png){ width=50% }
+
+## Fourier Series
+The properties just discussed for the sine series also apply to the cosine series.
+Putting them together forms the full Fourier series
+$$
+  f(x) = a_0 + \sum_n^\infty a_n \cos\left(\frac{n \pi x}{l}\right)
+  + \sum_n^\infty b_n \sin\left(\frac{n \pi x}{l}\right) \text{, }
+  -l < x < l \text{.}
+$$
+These functions can be expressed compactly in complex form, using the fact that $e^{i \theta} = \cos\theta + i \sin\theta$,
+$$
+  f(x) = \sum_n^\infty c_n e^{i n \pi x / l}\text{, } c_n \in \mathbb{C} \text{.}
+$$
+
+## Fourier Transform
+By construction, the Fourier series we defined above is periodic on an interval
+of $2l$. Evaluating it outside $-l < x < l$, we get copies of the same function.
+
+![](./plots/fft_ex_fun2_period.png){ width=50% }
+
+## Fourier Transform
+What if we want to decompose a function that is not periodic?
+We can extend the domain to be arbitrarily large:
+$$
+l \rightarrow \infty \text{,} \quad \frac{n}{2 l} \rightarrow k \text{,}
+$$
+where $k$ is the continuous limit of $n$, the "frequency" of the sine function.
+The sum over $n$ becomes an integral over $k$, and the coefficients $c_n$, a
+continuous function of $k$
+$$
+  f(x) = \int_{-\infty}^{\infty} \frac{dk}{2 \pi}~e^{2 \pi i k x} F(k) \text{.}
+$$
+
+## Fourier Transform
+$F(k)$ is the Fourier transform of $f(x)$. Because it is common to take Fourier
+transforms of time-series data, we often refer to $f(x)$ as being in the "time
+domain" as $F(k)$ its representation in the "frequency domain". There is no
+fundamental difference between the two, the inverse transform has the same form
+and is lossless.
+$$
+  F(k) = \int_{-\infty}^{\infty} dx~e^{-2 \pi i k x} f(x) \text{,}
+$$
+$$
+  f(x) = \int_{-\infty}^{\infty} \frac{dk}{2 \pi}~e^{2 \pi i k x} F(k) \text{.}
```

Comment:
With frequencies rather than angular frequencies, there isn't the 2 pi in the denominator.